File: src/lib/guest-contact.ts
import { updateAccountProfile } from "@/lib/profile-browser";
import { normalizeTenDigitPhone } from "@/lib/phone";

export type GuestContactDetails = {
  fullName: string;
  email: string;
  phone: string;
};

export function validateGuestContact(contact: GuestContactDetails): string | null {
  const fullName = contact.fullName.trim();
  if (fullName.length < 2) {
    return "Enter your full name.";
  }

  const email = contact.email.trim();
  if (!email || !/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)) {
    return "Enter a valid email address.";
  }

  if (!normalizeTenDigitPhone(contact.phone)) {
    return "Enter a valid 10-digit mobile number.";
  }

  return null;
}

export async function syncGuestContactDetails(contact: GuestContactDetails): Promise<void> {
  const validationError = validateGuestContact(contact);
  if (validationError) {
    throw new Error(validationError);
  }

  await updateAccountProfile({
    fullName: contact.fullName.trim(),
    phone: normalizeTenDigitPhone(contact.phone)!,
  });
}

File: backend/app/services/guest_contact.py
"""Shared guest contact validation before booking."""

from __future__ import annotations


def assert_guest_contact_ready(profile: dict, user) -> None:
    full_name = (profile.get("full_name") or "").strip()
    phone = (profile.get("phone") or "").strip()
    email = (getattr(user, "email", None) or "").strip()

    if not full_name:
        raise ValueError("Full name is required. Add your name before booking.")
    if not phone:
        raise ValueError("Phone number is required. Add your mobile number before booking.")
    if not email:
        raise ValueError("Email is required. Add your email before booking.")
